### PR TITLE
Encapsulated data loading functions in tactile pushing example into a new class

### PR DESCRIPTION
**Pull request by luisenp**
## Motivation and Context
This is one of several coming PRs In preparation for the backward pass experiments. I'm cleaning up the tactile pushing example to make it easier to run the resulting optimization problems in scripts outside of the examples folder.

## How Has This Been Tested
Ran 1 epoch of the example and the results match the original.

image

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/tactile_pose_estimation.py [hunk 1]
```diff
@@ -281,18 +275,11 @@ def run_learning_loop(cfg):
 
     outer_optim = optim.Adam(learnable_params, lr=learning_rate)
     batch_size = cfg.train.batch_size
-    num_batches = cfg.train.num_batches
-    measurements = [
-        (
-            dataset["img_feats"][0:time_steps].unsqueeze(0).repeat(batch_size, 1, 1),
```
Comment (luisenp) [inline examples/tactile_pose_estimation.py:287]: @psodhi Maybe you had already explained this and I forgot, but I'm curious about why this repeats the same trajectory data for each batch element and batch. Is this a placeholder? 

If I wanted to change this to select different trajectory data, what changes should I make?

File: examples/tactile_pose_estimation.py [hunk 2]
```diff
@@ -281,18 +275,11 @@ def run_learning_loop(cfg):
...
-        )
-        for _ in range(num_batches)
-    ]
-
-    obj_poses_gt = dataset["obj_poses"][0:time_steps, :].clone().requires_grad_(True)
-    eff_poses_gt = dataset["eff_poses"][0:time_steps, :].clone().requires_grad_(True)
+    measurements = dataset.get_measurements(
+        cfg.train.batch_size, cfg.train.num_batches, time_steps
+    )
+    obj_poses_gt = dataset.obj_poses[0:time_steps, :].clone().requires_grad_(True)
```
Comment (luisenp) [inline examples/tactile_pose_estimation.py:281]: @psodhi Another question. Is setting `requires_grad_(True)` necessary here?

**Review by mhmukadam — APPROVED** [review]
Looks great, nicely streamlined data loading!

File: examples/tactile_pose_estimation.py (hunk 2 above)
Comment (psodhi) [reply examples/tactile_pose_estimation.py:281]: I don't think it is necessary. We can remove it.

File: examples/tactile_pose_estimation.py (hunk 1 above)
Comment (psodhi) [reply examples/tactile_pose_estimation.py:287]: Yes, it is a placeholder. It is currently repeating a single trajectory. To select different trajectories, you would have to pass in different episodes indexes (`cfg.episode`) in `dataset = theg.TactilePushingDataset(dataset_path, sdf_path, cfg.episode, device).` And concatenate the outputs to get a batch dataset. Note that different `cfg.episode` trajectories can have different lengths, so it may require some min length trimming / padding.
